### PR TITLE
Adding Getting Started Page to SmartX documentation

### PR DESCRIPTION
**Pull request by seanmalloy-crest**
* Created more documentation with SmartX
* Improved the Readme file for SmartX
* Helped remedy some spelling mistakes and made some updates on the documentation
* Added a Sphinx capability to have dropdowns, so that long code examples can be neatly tucked into the documentation.

### CODE REVIEW

File: pytest_splunk_addon_ui_smartx/components/entity.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ class Entity(BaseComponent):
     def __init__(self, browser, container, add_btn=None):
         """
             :param browser: The selenium webdriver
-            :param container: Container in which the table is located. Of type dictionary: {"by":..., "select":...}
+            :param container: Container in which the emtity is located. Of type dictionary: {"by":..., "select":...}
```
Comment (jayjoshi-crest) [inline pytest_splunk_addon_ui_smartx/components/entity.py:25]: typo: Entity

**Review by jayjoshi-crest — APPROVED** [review]

File: pytest_splunk_addon_ui_smartx/components/entity.py (hunk 1 above)
Comment (seanmalloy-crest) [reply pytest_splunk_addon_ui_smartx/components/entity.py]: Typo fixed here: b99b13d

**Pull request merged**